src/components/MapArea.js

```diff
@@ -88,13 +91,24 @@ export default function MapArea() {
       ...DEFAULT_VIEWPORT,
     });
 
+    // Persist data when switching base layers (streets, satellites, etc)
     map.current.on("style.load", () => {
-      displayDataOnMap(map, trailData);
+      displayDataOnMap(map, selectedTrail)
     });
 
     return () => map.current.remove();
   }, []);
 
+  // Render trail data based on the trail selected in the sidebar
+  useEffect(() => {
+    if (map && map.current.getLayer("trailData")) {
```

Comment:
we can take advantage of optional chaining here:
```suggestion
    if (map?.current.getLayer("trailData")) {
```